dartagnan/src/main/java/com/dat3m/dartagnan/encoding/ProgramEncoder.java

```diff
@@ -156,7 +156,7 @@ public BooleanFormula encodeMemory() {
         // For all objects, their 'final' value fetched here represents their constant value.
         final var addrExprs = new ArrayList<BooleanFormula>();
         for (final MemoryObject object : memory.getObjects()) {
-            final BigInteger addressInteger = object.getValue();
```

Comment:
How was this never causing problems in the past?

Reply:
`getValue` and `getAddress` returned the same value.

Btw. you should try to mark both the old and the new code (and enough surrounding context), otherwise I need to look at the file to see what has changed.